disco_aws_automation/disco_metanetwork.py

```diff
@@ -32,13 +32,35 @@ class DiscoMetaNetwork(object):
     Representation of a disco meta-network. Contains a subnet for each availability zone,
     along with a route table which is applied all the subnets.
     """
-    def __init__(self, name, vpc):
+    def __init__(self, name, vpc, network_cidr=None):
         self.vpc = vpc
         self.name = name
+        if network_cidr:
+            self._network_cidr = IPNetwork(network_cidr)
         self._route_table = None  # lazily initialized
         self._security_group = None  # lazily initialized
         self._subnets = None  # lazily initialized
 
+    @property
+    def network_cidr(self):
+        """Get the network_cidr for the meta network"""
+        if not self._network_cidr:
+            # if we don't have a network_cidr yet (if it wasn't passed in the constructor)
+            # then calculate it from the subnets
+            subnets = self._find_subnets()
```

Comment:
Just a heads up, I deleted the `_find_subnets` method in my NAT branch. You would need to use the `_create_subnets` method instead. And also I made a `subnets` a dictionary with the AZ names being the keys and the subnets being the values.